src/cmap/options.rs

```diff
@@ -94,23 +93,42 @@ pub struct ConnectionPoolOptions {
     #[serde(rename = "waitQueueTimeoutMS")]
     #[serde(default)]
     #[serde(deserialize_with = "bson_util::deserialize_duration_from_u64_millis")]
-    pub wait_queue_timeout: Option<Duration>,
+    pub(crate) wait_queue_timeout: Option<Duration>,
 }
 
 impl ConnectionPoolOptions {
     pub(crate) fn from_client_options(options: &ClientOptions) -> Self {
-        let mut pool_options = Self::builder().build();
-        pool_options.app_name = options.app_name.clone();
-        pool_options.connect_timeout = options.connect_timeout;
-        pool_options.credential = options.credential.clone();
-        pool_options.driver_info = options.driver_info.clone();
-        pool_options.event_handler = options.cmap_event_handler.clone();
-        pool_options.max_idle_time = options.max_idle_time;
-        pool_options.min_pool_size = options.min_pool_size;
-        pool_options.tls_options = options.tls_options();
-        pool_options.wait_queue_timeout = options.wait_queue_timeout;
-
-        pool_options
+        Self {
```

Comment:
Using the struct literal here ensures that no fields can accidentally be forgotten (e.g. like `server_api` was)